swaybar/ipc.c

```diff
@@ -298,6 +304,21 @@ bool ipc_get_workspaces(struct swaybar *bar) {
 					calloc(1, sizeof(struct swaybar_workspace));
 				ws->num = json_object_get_int(num);
 				ws->name = strdup(json_object_get_string(name));
+				ws->label = strdup(ws->name);
+				char *ws_dup = strdup(ws->name);
+				char *stripped_name = strtok(ws_dup, ":");
+				size_t num_len = strlen(stripped_name) + 1;
+				// ws->num will be 0 if workspace name doesn't begin with int.
+				if (stripped_name != NULL && 
+						strlen(ws->name) > num_len && ws->num != -1) {
+					if(bar->config->strip_workspace_name) {
+						ws->label = strdup(stripped_name);
```

Comment:
You need to free `ws->label` before assigning a new value here to avoid a memory leak.

Reply:
Ah, so I need to free the pointer before reassign otherwise it will be dangling, correct?

Reply:
Correct. `ws->label` held the only reference to the value.

Reply:
Got it, thanks.